npm/pkg/dataplane/dataplane-test-cases_windows_test.go

```diff
@@ -0,0 +1,448 @@
+package dataplane
+
+import (
+	"github.com/Azure/azure-container-networking/network/hnswrapper"
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane/ipsets"
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane/policies"
+	dptestutils "github.com/Azure/azure-container-networking/npm/pkg/dataplane/testutils"
+	"github.com/Microsoft/hcsshim/hcn"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// tags
+const (
+	podCrudTag    Tag = "pod-crud"
+	nsCrudTag     Tag = "namespace-crud"
+	netpolCrudTag Tag = "netpol-crud"
+)
+
+const (
+	thisNode  = "this-node"
+	otherNode = "other-node"
+
+	ip1 = "10.0.0.1"
+	ip2 = "10.0.0.2"
+
+	endpoint1 = "test1"
+	endpoint2 = "test2"
+)
+
+// IPSet constants
+var (
+	podK1Set   = ipsets.NewIPSetMetadata("k1", ipsets.KeyLabelOfPod)
+	podK1V1Set = ipsets.NewIPSetMetadata("k1:v1", ipsets.KeyValueLabelOfPod)
+	podK2Set   = ipsets.NewIPSetMetadata("k2", ipsets.KeyLabelOfPod)
+	podK2V2Set = ipsets.NewIPSetMetadata("k2:v2", ipsets.KeyValueLabelOfPod)
+
+	// emptySet is a member of a list if enabled in the dp Config
+	// in Windows, this Config option is actually forced to be enabled in NewDataPlane()
+	emptySet      = ipsets.NewIPSetMetadata("emptyhashset", ipsets.EmptyHashSet)
+	allNamespaces = ipsets.NewIPSetMetadata("all-namespaces", ipsets.KeyLabelOfNamespace)
+	nsXSet        = ipsets.NewIPSetMetadata("x", ipsets.Namespace)
+	nsYSet        = ipsets.NewIPSetMetadata("y", ipsets.Namespace)
+
+	nsK1Set   = ipsets.NewIPSetMetadata("k1", ipsets.KeyLabelOfNamespace)
+	nsK1V1Set = ipsets.NewIPSetMetadata("k1:v1", ipsets.KeyValueLabelOfNamespace)
+	nsK2Set   = ipsets.NewIPSetMetadata("k2", ipsets.KeyLabelOfNamespace)
+	nsK2V2Set = ipsets.NewIPSetMetadata("k2:v2", ipsets.KeyValueLabelOfNamespace)
+)
+
+// DP Configs
+var (
+	defaultWindowsDPCfg = &Config{
+		IPSetManagerCfg: &ipsets.IPSetManagerCfg{
+			IPSetMode:          ipsets.ApplyAllIPSets,
+			AddEmptySetToLists: true,
+		},
+		PolicyManagerCfg: &policies.PolicyManagerCfg{
+			PolicyMode: policies.IPSetPolicyMode,
+		},
+	}
+)
+
+func policyXBaseOnK1V1() *networkingv1.NetworkPolicy {
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "base",
+			Namespace: "x",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"k1": "v1",
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{},
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+				networkingv1.PolicyTypeEgress,
+			},
+		},
+	}
+}
+
+func getAllSerialTests() []*SerialTestCase {
```

Comment:
Do we have any way of injecting expected errors to model nonhappy paths? Suppose we wanted to inject dp/hns returning errors and getting coverage in the controllers around those scenarios